### DISCUSSION TITLE
use inequality sign in Yalmip

### DISCUSSION THREAD

**Discussion by bioinfor-123** [yalmip/YALMIP#1333] [General discussion] [answered]
how to model "x ~= 3" as a constraint in Yalmip ?

**Comment by johanlofberg** [comment, marked as answer]
Although ~= is supported, you should write using a disjunction using two binary variable a and b so you understand what you actually model

`[implies(a,x>=4), implies(b, x<= 2) , a+b == 1]`

Another way is to write it using two integers y and z

`[x == 3 + y - z, y>=0, z>=0, y+z>=1]`

**Reply by bioinfor-123** [reply]
if x is a real number, not an integer, how to model "x ~= 3" as a constraint in Yalmip ?

**Reply by johanlofberg** [reply]
`[implies(a,x>=this_is_ok), implies(b, x<= this_is_also_ok) , a+b == 1]`

**Comment by kaiwangbeijingdehuoche** [comment]


**Reply by johanlofberg** [reply]
Don't hijack unrelated threads for new questions.

I don't see any specifically strange in your contraints, https://yalmip.github.io/tutorial/basics/
